Assets/Scripts/UI/Tooltip.cs

```diff
@@ -0,0 +1,221 @@
+        public Texture image;
+
+        // Reference Components
+        private Renderer render;
+        private Material[] allMaterials;
+        private Color[] baseColor;
+        private float t = 0f;
+        private bool over = false;
+        private string currentText = "";
+        private GUIStyle tooltipStyle = new GUIStyle();
+        private GUIStyle tooltipStyleShadow = new GUIStyle();
+        private Vector3 positionToScreen;
+        private float cameraDistance;
+
+
+        // Initialization
+        private void Start()
+        {
+            // Get all materials and all colors for supporting multi-materials object
+            render = GetComponent<Renderer>();
+            allMaterials = render.materials;
+            baseColor = new Color[allMaterials.Length];
+            int temp_length = baseColor.Length;
+            for (int i = 0; i < temp_length; i++)
+            {
+                baseColor[i] = allMaterials[i].color;
+            }
+
+            // Start settings of the tooltip
+            if (showTooltip)
+            { // Tooltip text style customization
+                if (tooltipUIPanel != null)
+                { // Initialization of the UI Panel
+                    tooltipUIPanel.SetActive(false);
+                }
+                tooltipStyle.normal.textColor = tooltipColor; // Color of the tooltip text
+                tooltipStyleShadow.normal.textColor = tooltipShadowColor; // Color of the tooltip shadow
+                tooltipStyle.fontSize = tooltipStyleShadow.fontSize = tooltipSize; // Size of the tooltip font
+                tooltipStyle.fontStyle = tooltipStyleShadow.fontStyle = FontStyle.Bold; // Style of the tooltip font
+                tooltipStyle.font = tooltipStyleShadow.font = tooltipFont;
+                switch (tooltipAlignment)
+                { // Alignment of the tooltip text
+                    case TooltipAlignment.Center:
+                        tooltipStyle.alignment = tooltipStyleShadow.alignment = TextAnchor.UpperCenter;
+                        break;
+                    case TooltipAlignment.Left:
+                        tooltipStyle.alignment = tooltipStyleShadow.alignment = TextAnchor.UpperLeft;
+                        break;
+                    case TooltipAlignment.Right:
+                        tooltipStyle.alignment = tooltipStyleShadow.alignment = TextAnchor.UpperRight;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        // Update once per frame
+        private void Update()
+        {
+            if (over && t < 1)
+            { // Fade in hightlight color
+                foreach (Material material in allMaterials)
+                {
+                    material.color = Color.Lerp(material.color, highlightColor, t);
+                }
+                t += highlightSpeed * Time.deltaTime;
+            }
+            else if (!over && t < 1f)
```

Comment:
Will stop being triggered at any point except when the tooltip is showing?

Reply:
Hmm i don't think so. Maybe i will try take those and put in beginTooltip() and endTooltip() method

Reply:
Hmm i don't think it will. But actually it will only be triggered when mouse cursor enter or exit an object.